src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+    private static User user1;
+    private static User user2;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user1 = new User();
+        user1.setAge(28);
+        user1.setLogin("BorysenkoDanyl");
+        user1.setPassword("Danyl5555");
+        user2 = new User();
+        user2.setAge(25);
+        user2.setLogin("BorysenkoDanyl");
+        user2.setPassword("Borysenko");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user1.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void register_nullAge_NotOk() {
+        user1.setAge(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        user1.setPassword(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        registrationService.register(user1);
+        assertTrue(Storage.people.contains(user1));
+    }
+
+    @Test
+    void register_existedUser_NotOk() {

```

Comment:
Why do you use another approach in this method?
![image](https://user-images.githubusercontent.com/20326858/205598170-ff53cc43-8bf6-4879-b85f-ca1e02f75c91.png)